Proposal for rootless Kubeflow

juliusvonkohout opened the pull request:
**Which issue is resolved by this Pull Request:**
Related to #2528 

**Description of your changes:**
Add proposal for rootless Kubeflow.

**Checklist:**
- [ ] Unit tests pass:
  **Make sure you have installed kustomize == 3.2.1**
    1. `make generate-changed-only`
    2. `make test`

annajung commented on proposals/20200913-rootlessKubeflow.md:
```diff
@@ -0,0 +1,35 @@
+We want to run Kubeflow 99 % rootless accoring to CNCF/Kubernetes best practices.
+
+The main steps are adding an additional profile for istio-cni or ambient mesh, updating the documentation and manifest generation process. 
+Then adding the baseline and restricted PSS as kustomize component to /contrib and extending the profile controller to annotate user namespaces with configurable PSS labels.
```
Could you expand more on why additional profile for istio-cni or ambient mesh is necessary in addition to the PSS set

annajung commented on proposals/20200913-rootlessKubeflow.md:
```diff
@@ -0,0 +1,35 @@
+We want to use a staged approach.
+First Stage:
+
+1. Implement Istio 1.17.5 and use it by default. This is important for the Kubeflow 1.8  feature freeze
```
Is it more appropriate to say istio version > 1.17? and can you specify why > 1.17

annajung commented on proposals/20200913-rootlessKubeflow.md:
```diff
@@ -0,0 +1,35 @@
+5. Add istio-ambient as an option to Kubeflow 1.9
+6. Enforce PSS baseline (here you can still build OCI containers via Podman and buildah) in Kubeflow 1.9. It works with any istio
+7. Warning for violations of restricted PSS
+8. Optionally Enforce PSS restricted (this is where minor corner cases are affected) in Kubeflow 1.10
```
As an additional step, I would like to see a modification to the test from the first step to test out not just installation but PSS feature

annajung commented on proposals/20200913-rootlessKubeflow.md:
```diff
@@ -0,0 +1,35 @@
+8. Optionally Enforce PSS restricted (this is where minor corner cases are affected) in Kubeflow 1.10
+
+Third stage:
+10. Upgrade Istio to 1.19 and use ambient service mesh by default in Kubeflow 1.10
```
Why specifically 1.19? Can we be more generic to say, upgrade istio? If 1.19 is needed, is it more appropriate to say istio > 1.19?

annajung reviewed (commented):
Thanks for getting started on this, I left a few comments to help drive out the proposal a little more to see specific details be added to the proposal

juliusvonkohout commented:
> Thanks for getting started on this, I left a few comments to help drive out the proposal a little more to see specific details be added to the proposal

i hope to have addressed all of your comments

kimwnasptd commented:
We discussed and reviewed this live in the Manifests WG meeting on Sept 21 of 2023 and worked a little bit on the structure. This is a really great proposal, great work @juliusvonkohout 

/lgtm
/approve
